extensions/functions_geometry.yaml

```diff
@@ -33,3 +33,103 @@ scalar_functions:
           - name: y
             value: u!geometry
         return: u!geometry
+  -
+    name: "x_coordinate"
+    description: >
+      Return the x coordinate of the point.  Return null if not available.
+    impls:
+      - args:
+          - name: point
+            value: u!geometry
+        return: fp64
+  -
+    name: "y_coodinate"
+    description: >
+      Return the y coordinate of the point.  Return null if not available.
+    impls:
+      - args:
+          - name: point
+            value: u!geometry
+        return: fp64
+  -
+    name: "numpoints"
+    description: >
+      Return the number of points in the geometry.  The geometry should be an linestring
+      or circularstring.
+    impls:
+      - args:
+          - name: geom
+            value: u!geometry
+        return: i64
+  -
+    name: "isempty"
+    description: >
+      Return true is the geometry is an empty geometry.
+    impls:
+      - args:
+          - name: geom
+            value: u!geometry
+        return: boolean
+  -
+    name: "isclosed"
+    description: >
+      Return true if the geometry's start and end points are the same.
+    impls:
+      - args:
+          - name: geom
+            value: geometry
+        return: boolean
+  -
+    name: "issimple"
+    description: >
+      Return true if the geometry does not self intersect.
+    impls:
+      - args:
+          - name: geom
+            value: u!geometry
+        return: boolean
+  -
+    name: "isring"
+    description: >
+      Return true if the geometry's start and end points are the same and it does not self
+      intersect.
+    impls:
+      - args:
+          - name: geom
+            value: u!geometry
+        return: boolean
+  -
+    name: "geometrytype"
+    description: >
+      Return the type of geometry as a string.
+    impls:
+      - args:
+          - name: geom
+            value: u!geometry
+        return: string
+  -
+    name: "envelope"
+    description: >
+      Return the minimum bounding box for the input geometry as a geometry.
+
+      The returned geometry is defined by the corner points of the bounding box.  If the
+      input geometry is a point or a line, the returned geometry can also be a point or line.
+    impls:
+      - args:
+          - name: geom
+            value: u!geometry
+        return: u!geometry
+  -
+    name: "dimension"
+    description: >
+      Return the dimension of the input geometry.  If the input is a collection of geometries,
```

Comment:
The examples below don't necessarily imply dimension.  Could we clarify if a line in 3D space considered one dimensional or three?

Reply:
I believe it would still be one dimensional.  I don't think this function takes into consideration the dimensionality of the space, only the geometry itself.  Maybe @paleolimbot would have a better idea on this.

Reply:
I think all we need to do is add a line that it refers to the complexity of the figures in the geometry and not the coordinate system being used.

Reply:
Sounds good.  Just updated.